liveblog.php

```diff
@@ -576,52 +711,88 @@ public static function ajax_lazyload_entries() {
-				'entries' => array(),
-				'index'   => (int) filter_input( INPUT_GET, 'index' ),
-			) );
+		self::json_return( $result_for_json );
+	}
+
+	/**
+	 * Get all Liveblog entries that are to be lazyloaded.
+	 *
+	 * @param int $max_timestamp Maximum timestamp for the Liveblog entries.
+	 * @param int $min_timestamp Minimum timestamp for the Liveblog entries.
+	 *
+	 * @return array An array of json encoded results
+	 */
+	public static function get_lazyload_entries( $max_timestamp, $min_timestamp ) {
+
+		if ( empty( self::$entry_query ) ) {
+			self::$entry_query = new WPCOM_Liveblog_Entry_Query( self::$post_id, self::key );
 		}
 
-		$entries = array_slice( $entries, 0, WPCOM_Liveblog_Lazyloader::get_number_of_entries() );
+		// Get all Liveblog entries that are to be lazyloaded.
+		$entries = self::$entry_query->get_for_lazyloading( $max_timestamp, $min_timestamp );
 
 		$entries_for_json = array();
 
-		// Set up an array containing the JSON data for all Liveblog entries.
-		foreach ( $entries as $entry ) {
-			$entries_for_json[] = $entry->for_json();
+		if ( ! empty( $entries ) ) {
+			$entries = array_slice( $entries, 0, WPCOM_Liveblog_Lazyloader::get_number_of_entries() );
+
+			// Populate an array containing the JSON data for all Liveblog entries.
+			foreach ( $entries as $entry ) {
+				$entries_for_json[] = $entry->for_json();
+			}
 		}
 
-		// Set up the data to be returned via JSON.
-		$result_for_json = array(
+		$result = array(
 			'entries' => $entries_for_json,
 			'index'   => (int) filter_input( INPUT_GET, 'index' ),
 		);
 
-		do_action( 'liveblog_entry_request', $result_for_json );
+		if ( ! empty( $entries_for_json ) ) {
+			do_action( 'liveblog_entry_request', $result );
+			self::$do_not_cache_response = true;
+		} else {
+			do_action( 'liveblog_entry_request_empty' );
+		}
 
-		self::json_return( $result_for_json );
+		//self::json_return( $result_for_json );
+
+		return $result;
 	}
 
 	public static function ajax_preview_entry() {
 		$entry_content = isset( $_REQUEST['entry_content'] ) ? $_REQUEST['entry_content'] : '';
+		$entry_content = self::format_preview_entry( $entry_content );
+
+		self::json_return( $entry_content );
+	}
+
+	/**
+	 * Format the passed in content and give it back in an array
+	 *
+	 * @param string $entry_content The entry content to be previewed
+	 *
+	 * @return array The entry content wrapped in HTML elements
+	 */
+	public static function format_preview_entry( $entry_content ) {
+
 		$entry_content = stripslashes( wp_filter_post_kses( $entry_content ) );
 		$entry_content = apply_filters( 'liveblog_before_preview_entry', array( 'content' => $entry_content ) );
 		$entry_content = $entry_content['content'];
 		$entry_content = WPCOM_Liveblog_Entry::render_content( $entry_content );
 
 		do_action( 'liveblog_preview_entry', $entry_content );
 
-		self::json_return(
-			array( 'html' => $entry_content ),
-			array( 'cache' => false )
-		);
+		// self::json_return(
```

Comment:
Code that isn't used or needed should be removed completely, rather than commenting it out. If it's needed in the future it can be restored via the Git history.